### ISSUE TITLE
MacOS docfx broken between 2.51 and 2.52

### ISSUE THREAD

**Issue by amirebrahimi** [dotnet/docfx#5785]
**Operation System**: (`Windows` or `Linux` or `MacOS`)
MacOS

**DocFX Version Used**: 
2.52

#5637 introduced a [new version](https://github.com/dotnet/docfx/commit/ffba70567a8157e659b78d6254d88ac8efeeec4a#diff-370fa3bbd2b361f76dff6127d96b19ceR30) of the FSharp compiler service (34.1.1), which seems to be incompatible with MacOS / mono fork of MSBuild (or does this now require an update to mono?). I used git bisect to track it down to this changeset: ffba7056.

The error received is:
```
20-04-16 03:21:06.424]Info:[MetadataCommand]Completed Scope:MetadataCommand in 705.8239 milliseconds.
[20-04-16 03:21:06.424]Error:Error extracting metadata for /Users/amir/Library/Application Support/unity/editor/documentation/build/com.unity.package-manager-doctools@1.1/package/solution.csproj: System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.SaveAllMembersFromCacheAsync () [0x00535] in <8a8478fd5e284acdb5322455028f571a>:0 
  at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.ExtractMetadataAsync () [0x00106] in <8a8478fd5e284acdb5322455028f571a>:0 
[20-04-16 03:21:06.424]Info:Completed in 709.2319 milliseconds
```
but this is simply because the project loader fails to load the project.

For now we'll have to stick with 2.51 until this is fixed.

**Comment by jo3w4rd** [comment]
Some additional information:

The previous release, 2.51 works, both for our own projects and for the DocFX Seed project. 2.52 does work for the same projects on Windows.

With 2.52 (downloaded as a zip from the releases tab), generating the docs for the DocFX Seed project shows warnings about the project files in the command line output and the results are missing any HTML for the C# API files. An example warning is, "... SampleClass1.sln: Could not resolve type with token 010000ed from typeref (expected class 'Microsoft.Build.Construction.SolutionFile' in assembly 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a')"

For our own projects, the command line output shows exceptions thrown relating to the project file and nothing is generated. First there is a warning, "Unity.Entities.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored."
This is followed later with a NULL reference exception (shown below).

The full DocFX Seed command line output:
`
charleswMBP13TB:dfxtest charlesw$ mono ../docfx-2.52/docfx.exe docfx-seed/docfx.json --serve
[20-04-16 02:20:12.776]Info:[MetadataCommand.ExtractMetadata]Using mono /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/15.0/bin/MSBuild.dll as inner compiler.
[20-04-16 02:20:12.902]Info:[MetadataCommand.ExtractMetadata]Loading projects...
[20-04-16 02:20:13.435]Warning:MetadataCommand.ExtractMetadataError opening solution /Users/charlesw/Documents/dfxtest/docfx-seed/src/SampleClass1/SampleClass1.sln: Could not resolve type with token 010000ed from typeref (expected class 'Microsoft.Build.Construction.SolutionFile' in assembly 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'). Ignored.
[20-04-16 02:20:13.454]Info:[MetadataCommand.ExtractMetadata]Cache for /Users/charlesw/Documents/dfxtest/docfx-seed/src/SampleClass1/SampleClass1.sln in /Users/charlesw/Documents/dfxtest/docfx-seed/src/SampleClass1/obj/xdoc/cache/final/645371506 is corrupted, rebuild...
[20-04-16 02:20:13.454]Info:[MetadataCommand.ExtractMetadata]Generating metadata for each project...
[20-04-16 02:20:13.460]Warning:[MetadataCommand.ExtractMetadata]No metadata is generated for .
[20-04-16 02:20:13.479]Info:[MetadataCommand]Completed Scope:MetadataCommand in 735.2815 milliseconds.
[20-04-16 02:20:13.608]Info:[BuildCommand]6 plug-in(s) loaded.
[20-04-16 02:20:13.613]Info:[BuildCommand]Post processor ExtractSearchIndex loaded.
[20-04-16 02:20:13.624]Info:[BuildCommand]No files are found with glob pattern **/*.yml, excluding , under directory "/Users/charlesw/Documents/dfxtest/docfx-seed/obj/api"
[20-04-16 02:20:13.651]Info:[BuildCommand]Markdown engine is markdig
[20-04-16 02:20:13.782]Info:[BuildCommand.BuildCore.Build Document]Max parallelism is 4.
[20-04-16 02:20:14.341]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.TocDocumentProcessor]Building 3 file(s) in TocDocumentProcessor(BuildTocDocument)...
[20-04-16 02:20:14.341]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ResourceDocumentProcessor]Building 1 file(s) in ResourceDocumentProcessor(ValidateResourceMetadata)...
[20-04-16 02:20:14.341]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ConceptualDocumentProcessor]Building 0 file(s) in ConceptualDocumentProcessor(BuildConceptualDocument=>CountWord=>ValidateConceptualDocumentMetadata)...
[20-04-16 02:20:14.341]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.RestApiDocumentProcessor]Building 10 file(s) in RestApiDocumentProcessor(BuildRestApiDocument=>ValidateRestApiDocumentMetadata=>ApplyOverwriteDocumentForRestApi)...
[20-04-16 02:20:14.355]Warning:BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.TocDocumentProcessor.Prebuild.BuildTocDocumentUnable to find either toc.yml or toc.md inside obj/api/. Make sure the file is included in config file docfx.json!
[20-04-16 02:20:15.009]Info:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.Apply Templates]Applying templates to 6 model(s)...
[20-04-16 02:20:15.257]Info:[BuildCommand.BuildCore.Build Document]XRef map exported.
[20-04-16 02:20:15.380]Info:[BuildCommand.Postprocess.HandlePostProcessorsWithIncremental.HandlePostProcessors.Processing ExtractSearchIndex]Extracting index data from 7 html files
[20-04-16 02:20:15.474]Info:[BuildCommand.Postprocess]Manifest file saved to manifest.json.
[20-04-16 02:20:15.499]Info:[BuildCommand]Completed building documents in 1885.5855 milliseconds.
Serving "/Users/charlesw/Documents/dfxtest/docfx-seed/_site" on http://localhost:8080

[20-04-16 02:24:02.679]Info:[BuildCommand]Completed Scope:BuildCommand in 229199.6742 milliseconds.
[20-04-16 02:24:02.679]Info:Completed in 229937.8832 milliseconds

Build succeeded with warning.
[20-04-16 02:24:02.684]Warning:MetadataCommand.ExtractMetadataError opening solution /Users/charlesw/Documents/dfxtest/docfx-seed/src/SampleClass1/SampleClass1.sln: Could not resolve type with token 010000ed from typeref (expected class 'Microsoft.Build.Construction.SolutionFile' in assembly 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'). Ignored.
[20-04-16 02:24:02.684]Warning:[MetadataCommand.ExtractMetadata]No metadata is generated for .
[20-04-16 02:24:02.684]Warning:BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.TocDocumentProcessor.Prebuild.BuildTocDocumentUnable to find either toc.yml or toc.md inside obj/api/. Make sure the file is included in config file docfx.json!
3 Warning(s)
0 Error(s)
`

The full output from one of our projects:

`
charleswMBP13TB:Samples charlesw$ mono ../../docfx-2.52/docfx.exe docs/docfx_toc_enabled.json --serve
[20-04-15 11:38:15.482]Info:[MetadataCommand.ExtractMetadata]Using mono /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/15.0/bin/MSBuild.dll as inner compiler.
[20-04-15 11:38:15.784]Info:[MetadataCommand.ExtractMetadata]Loading projects...
[20-04-15 11:38:16.702]Warning:MetadataCommand.ExtractMetadataError opening project /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-04-15 11:38:16.709]Info:[MetadataCommand]Completed Scope:MetadataCommand in 1291.8044 milliseconds.
[20-04-15 11:38:16.709]Error:Error extracting metadata for /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: System.NullReferenceException: Object reference not set to an instance of an object
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.SaveAllMembersFromCacheAsync () [0x00380] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.ExtractMetadataAsync () [0x000c0] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
[20-04-15 11:38:16.709]Info:Completed in 1295.6479 milliseconds

Build failed.
[20-04-15 11:38:16.713]Warning:MetadataCommand.ExtractMetadataError opening project /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-04-15 11:38:16.713]Error:Error extracting metadata for /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: System.NullReferenceException: Object reference not set to an instance of an object
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.SaveAllMembersFromCacheAsync () [0x00380] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.ExtractMetadataAsync () [0x000c0] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
1 Warning(s)
1 Error(s)
charleswMBP13TB:Samples charlesw$ mono ../../docfx-2.52/docfx.exe docs/docfx_toc_enabled.json
[20-04-16 02:09:11.944]Info:[MetadataCommand.ExtractMetadata]Using mono /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/15.0/bin/MSBuild.dll as inner compiler.
[20-04-16 02:09:12.360]Info:[MetadataCommand.ExtractMetadata]Loading projects...
[20-04-16 02:09:13.441]Warning:MetadataCommand.ExtractMetadataError opening project /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-04-16 02:09:13.448]Info:[MetadataCommand]Completed Scope:MetadataCommand in 1583.7488 milliseconds.
[20-04-16 02:09:13.448]Error:Error extracting metadata for /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: System.NullReferenceException: Object reference not set to an instance of an object
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.SaveAllMembersFromCacheAsync () [0x00380] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.ExtractMetadataAsync () [0x000c0] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
[20-04-16 02:09:13.448]Info:Completed in 1587.8387 milliseconds

Build failed.
[20-04-16 02:09:13.451]Warning:MetadataCommand.ExtractMetadataError opening project /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-04-16 02:09:13.452]Error:Error extracting metadata for /Users/charlesw/Documents/dots/Samples/Unity.Entities.csproj: System.NullReferenceException: Object reference not set to an instance of an object
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.SaveAllMembersFromCacheAsync () [0x00380] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.ExtractMetadataAsync () [0x000c0] in <36cbd452dc6f4b5d8daa3dce4c7355a8>:0
1 Warning(s)
1 Error(s)

`

**Comment by awegg** [comment]
I am seeing the same on Linux with version 2.52. A downgrade to 2.51 also solved it for me.

**Comment by varunmankal** [comment]
@awegg how to install docfx 2.51 version ?

**Comment by RobinPerris** [comment]
In our current netstandard2.0 project we're unable to generate docs on macOS with any version of DocFX past 2.51. 

2.52-2.55 work just fine on Windows for the same project, but not macOS.

Currently running 2.51 by installing from the NuGet feed:
```
nuget install docfx.console -version 2.51.0
```

**Comment by nickwinder** [comment]
We also saw this issue in our `netstandard2.0` project on macOS. To solve we downgraded to docfx 2.51 using brew. 
```
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/2b977129bdd108f234ef20f2120103d46e4f2a6e/Formula/docfx.rb
```

**Comment by goda** [comment]
Had the same issue on macOS 10.15.5 with docfx 2.56 and mono 6.8.0 installed. After downgrading to docfx 2.51 everything worked. 

Thanks!

**Comment by halomademeapc** [comment]
I experienced this issue as well running docfx in Travis CI.  Setting the version to 2.51 works.  
https://github.com/halomademeapc/GravyIrc/blob/master/.travis.yml#L9

**Comment by DerBerndH** [comment]
Came across the issue with MacOS 10.15.6 an using mono 6.10.0.104. Also, downgrading to docfx 2.51 made it work. As this should also work on our build agents, fixing this in the upcoming version is very much appreciated. Installing/downgrading to an older version with Homebrew is not funny. Looking forward to have this working again.
Thanks!

**Comment by superyyrrzz** [comment]
@v-pegao to help investigate

AB#302725

**Comment by perlun** [comment]
Seeing the same problem with mono 6.12.0.90-0xamarin1+debian10b1 on Debian:

## docfx 2.56.2.0

```
$ ./docfx/docfx.exe --version
docfx 2.56.2.0
Copyright (C) 2020 © Microsoft Corporation. All rights reserved.
This is open-source software under MIT License.

$ ./docfx/docfx.exe docfx.json  --serve
[20-09-22 08:26:22.960]Info:[MetadataCommand.ExtractMetadata]Using mono /usr/lib/mono/msbuild/15.0/bin/MSBuild.dll as inner compiler.
[20-09-22 08:26:23.342]Info:[MetadataCommand.ExtractMetadata]Loading projects...
[20-09-22 08:26:24.906]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Common/Perlang.Common.csproj)Error opening project /home/per/git/perlang/Perlang.Common/Perlang.Common.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.908]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Tests/Perlang.Tests.csproj)Error opening project /home/per/git/perlang/Perlang.Tests/Perlang.Tests.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.908]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Parser/Perlang.Parser.csproj)Error opening project /home/per/git/perlang/Perlang.Parser/Perlang.Parser.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.909]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Tests.Integration/Perlang.Tests.Integration.csproj)Error opening project /home/per/git/perlang/Perlang.Tests.Integration/Perlang.Tests.Integration.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.910]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Stdlib/Perlang.Stdlib.csproj)Error opening project /home/per/git/perlang/Perlang.Stdlib/Perlang.Stdlib.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.910]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.ConsoleApp/Perlang.ConsoleApp.csproj)Error opening project /home/per/git/perlang/Perlang.ConsoleApp/Perlang.ConsoleApp.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.911]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Interpreter/Perlang.Interpreter.csproj)Error opening project /home/per/git/perlang/Perlang.Interpreter/Perlang.Interpreter.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.911]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/test/test.csproj)Error opening project /home/per/git/perlang/test/test.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.932]Info:[MetadataCommand]Completed Scope:MetadataCommand in 2067.7467 milliseconds.
[20-09-22 08:26:24.933]Error:Error extracting metadata for /home/per/git/perlang/Perlang.Common/Perlang.Common.csproj,/home/per/git/perlang/Perlang.Tests/Perlang.Tests.csproj,/home/per/git/perlang/Perlang.Parser/Perlang.Parser.csproj,/home/per/git/perlang/Perlang.Tests.Integration/Perlang.Tests.Integration.csproj,/home/per/git/perlang/Perlang.Stdlib/Perlang.Stdlib.csproj,/home/per/git/perlang/Perlang.ConsoleApp/Perlang.ConsoleApp.csproj,/home/per/git/perlang/Perlang.Interpreter/Perlang.Interpreter.csproj,/home/per/git/perlang/test/test.csproj: System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.SaveAllMembersFromCacheAsync () [0x00380] in <0779a76130d64c11baa2c8c084cbfe7c>:0 
  at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.ExtractMetadataAsync () [0x000c0] in <0779a76130d64c11baa2c8c084cbfe7c>:0 
[20-09-22 08:26:24.933]Info:Completed in 2079.7035 milliseconds


Build failed.
[20-09-22 08:26:24.945]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Common/Perlang.Common.csproj)Error opening project /home/per/git/perlang/Perlang.Common/Perlang.Common.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.946]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Tests/Perlang.Tests.csproj)Error opening project /home/per/git/perlang/Perlang.Tests/Perlang.Tests.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.946]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Parser/Perlang.Parser.csproj)Error opening project /home/per/git/perlang/Perlang.Parser/Perlang.Parser.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.946]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Tests.Integration/Perlang.Tests.Integration.csproj)Error opening project /home/per/git/perlang/Perlang.Tests.Integration/Perlang.Tests.Integration.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.946]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Stdlib/Perlang.Stdlib.csproj)Error opening project /home/per/git/perlang/Perlang.Stdlib/Perlang.Stdlib.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.947]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.ConsoleApp/Perlang.ConsoleApp.csproj)Error opening project /home/per/git/perlang/Perlang.ConsoleApp/Perlang.ConsoleApp.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.947]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/Perlang.Interpreter/Perlang.Interpreter.csproj)Error opening project /home/per/git/perlang/Perlang.Interpreter/Perlang.Interpreter.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.947]Warning:[MetadataCommand.ExtractMetadata](/home/per/git/perlang/test/test.csproj)Error opening project /home/per/git/perlang/test/test.csproj: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception.. Ignored.
[20-09-22 08:26:24.947]Error:Error extracting metadata for /home/per/git/perlang/Perlang.Common/Perlang.Common.csproj,/home/per/git/perlang/Perlang.Tests/Perlang.Tests.csproj,/home/per/git/perlang/Perlang.Parser/Perlang.Parser.csproj,/home/per/git/perlang/Perlang.Tests.Integration/Perlang.Tests.Integration.csproj,/home/per/git/perlang/Perlang.Stdlib/Perlang.Stdlib.csproj,/home/per/git/perlang/Perlang.ConsoleApp/Perlang.ConsoleApp.csproj,/home/per/git/perlang/Perlang.Interpreter/Perlang.Interpreter.csproj,/home/per/git/perlang/test/test.csproj: System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.SaveAllMembersFromCacheAsync () [0x00380] in <0779a76130d64c11baa2c8c084cbfe7c>:0 
  at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.ExtractMetadataAsync () [0x000c0] in <0779a76130d64c11baa2c8c084cbfe7c>:0 
	8 Warning(s)
	1 Error(s)
```

As for others, docfx 2.51.0.0 works better here too; I get some exceptions there as well (`Could not resolve type with token 01000069 from typeref (expected class 'NuGet.Packaging.Core.PackageDependency' in assembly 'NuGet.Packaging, Version=5.6.0.5, Culture=neutral, PublicKeyToken=31bf3856ad364e35')` and others), but at least the site seems to build correctly, extracting my XML documentation from the sources.

**Comment by burntsugar** [comment]
Seeing same issue on Mono 6.10.0.104 with docfx 2.56.2.0

**Comment by jeroenlandheer** [comment]
Same issue on Debian with Mono.

**Comment by panosru** [comment]
I bumped on the same issue today,  and from what I can tell I'm not alone... I also downgraded to v2.51.

For those who are getting the message `Error: Calling Installation of docfx from a GitHub commit URL is disabled! Use 'brew extract docfx' to stable tap on GitHub instead.` when running `brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/2b977129bdd108f234ef20f2120103d46e4f2a6e/Formula/docfx.rb` here is how I did it:

![image](https://user-images.githubusercontent.com/400362/95664931-78655f80-0b54-11eb-8942-e4796b358c77.png)

The reason I used the `--ignore-dependencies` flag is that I'm not using brew's mono (which is also outdated), I'm using the one provided by Visual Studio 2019 for Mac (even though I'm using Rider as my primary IDE).

Let's hope that the issue will be sorted out in future releases.

Cheers

**Comment by nickwinder** [comment]
@v-pegao 👍 That seems to work in my instance. Running as `mono docfx.exe my-docfx.json`.

**Comment by panosru** [comment]
@v-pegao I can also confirm what @nickwinder said, just tried on macOS:

![image](https://user-images.githubusercontent.com/400362/96265318-f51d9100-0fcd-11eb-8c86-ab7404c9ab9f.png)

**Comment by RobinPerris** [comment]
@v-pegao This fixes the issue I mentioned above - https://github.com/dotnet/docfx/issues/5785#issuecomment-646103314 👍

**Comment by superyyrrzz** [comment]
fixed in 2.56.4
